**Importing Libraries**

In [1]:
import pandas as pd  #data processing
import numpy as np  # linear algebra

 **Load the Data**

In [2]:
data = pd.read_csv('kc_house_data.csv')

 **First 5 rows of data**
 
 Since we have loaded the data, now we will read first 5 records of our data.

In [19]:
data.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

**Dropping of the particular column value**

We don't require the column id and date at this point so we will be dropping them from the dataset.

In [20]:
data = data.drop(['id','date'],axis=1)

**View the modified dataset**

In [21]:
data.head()

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  \
0  221900.0         3       1.00         1180      5650     1.0           0   
1  538000.0         3       2.25         2570      7242     2.0           0   
2  180000.0         2       1.00          770     10000     1.0           0   
3  604000.0         4       3.00         1960      5000     1.0           0   
4  510000.0         3       2.00         1680      8080     1.0           0   

   view  condition  grade  sqft_above  sqft_basement  yr_built  yr_renovated  \
0     0          3      7        1180              0      1955             0   
1     0          3      7        2170            400      1951          1991   
2     0          3      6         770              0      1933             0   
3     0          5      7        1050            910      1965             0   
4     0          3      8        1680              0      1987             0   

   zipcode      lat     long  sqft_living15  sqft_lot15  
0    98178  47.5112 -122.257           1340        5650  
1    98125  47.7210 -122.319           1690        7639  
2    98028  47.7379 -122.233           2720        8062  
3    98136  47.5208 -122.393           1360        5000  
4    98074  47.6168 -122.045           1800        7503

**Defining Independent and Dependent Variable**

In [22]:
# X(independent variable) & y(target variable)
X = data.iloc[:,1:].values
y = data.iloc[:,0].values

In [23]:
#Splitting the data into training and test dataset
from sklearn.model_selection import train_test_split

X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3,random_state=0)

In [24]:
len(X_train)

15129

In [25]:
len(X_test)

6484

 **Random Forest Regression model** 

In [30]:
from sklearn.ensemble import RandomForestRegressor

regressor = RandomForestRegressor(n_estimators=28, random_state=0)

regressor.fit(X_train,y_train) 

RandomForestRegressor(n_estimators=28, random_state=0)

In [31]:
 y_pred=regressor.predict(X_test)

In [32]:
y_pred

array([ 305641.07142857, 1555214.28571429,  526017.85714286, ...,
        349738.85714286,  228581.28571429,  195254.64285714])

In [33]:
result=pd.DataFrame({'Actual': y_test,'Predicted':y_pred})

In [34]:
result

Actual     Predicted
0      297000.0  3.056411e+05
1     1578000.0  1.555214e+06
2      562100.0  5.260179e+05
3      631500.0  5.330214e+05
4      780000.0  6.963723e+05
...         ...           ...
6479  1100000.0  8.738036e+05
6480   700000.0  6.632294e+05
6481   380000.0  3.497389e+05
6482   268000.0  2.285813e+05
6483   206000.0  1.952546e+05

[6484 rows x 2 columns]

**Accuracy of Model**

In [35]:
accuracy=100-np.mean(np.abs(y_test-y_pred)/np.abs(y_test))*100

In [36]:
accuracy

87.01172197472481

### Conclusion

From the above it is clear that random forest accuracy is 87%. So Random Forest is a suitable model for predicting the price of the house.